### DISCUSSION TITLE
v4.8.6

### DISCUSSION THREAD

**Discussion by marcelklehr** [floccusaddon/floccus#1006] [General]
### Fixes
 - build.gradle: Fix version
 - NewAccount: Link to importexport view for better discovery (only in browser)
 - [Android] Allow self-signed certificates added to the Android user cert store
 - [Native] Add FundDevelopment link target
 - [Native] Fix exit on back button
 - Account: Fix cancelSync
 - AccountCard: Remove indeterminate loading bar animation

<hr /><em>This discussion was created from the release <a href='https://github.com/floccusaddon/floccus/releases/tag/v4.8.6'>v4.8.6</a>.</em>

**Comment by misterhsp** [comment]
i cannot find the Option for self-signed cert in the App

...

**Reply by marcelklehr** [reply]
hi @misterhsp 

Floccus doesn't handle TLS encryption on its own, but relies on the Android system. Thus you will need to install your self-signed certificate on your Android system directly. [A search for "How to install a self-signed certificate on Android" yielded this page among others](https://proxyman.io/posts/2020-09-29-Install-And-Trust-Self-Signed-Certificate-On-Android-11) which may be helpful. Once your Android system trusts the self-signed certificate, floccus should, too.

Cheers

**Comment by misterhsp** [comment]
already done, it does not work.

...

**Reply by marcelklehr** [reply]
@elmodor says "Had to uninstall and install the app again for it to work." Have you tried this?

**Comment by misterhsp** [comment]
Yes, I have tried that too

...

**Comment by misterhsp** [comment]
Just as a side note: Android has not trusted user certificates for a long time. This is nothing new. I mean, it's been like that since Android 6. Look at the Nextcloud app, there is this option that it works with self-signed. That's what you need to put in your app.

Good luck...

**Comment by elmodor** [comment]
I'm using a caddy reverse proxy on my server and added the caddy root certificate (which is used for signing the ones for each site), to the android device as @marcelklehr already said.
Then it should work, as it did work with other apps as well. 

I think what you are talking about is to be able to manually add/trust certificates within the app which is probably more complex (because then it needs to handle TLS connection on its own)

If your server and android device is configured correctly, 4.8.6 will work with self signed certificates

**Comment by misterhsp** [comment]
It does not work! Not even with my own CA (roorcert)

...

**Comment by misterhsp** [comment]
I tried around some more, it just doesn't work.

...

**Reply by elmodor** [reply]
Not quite sure how to help you out. 
Are you trying to connect to a webdav server secured by SSL? Can you connect to the webdav server in your browser directly? (via https?)
You added the CA authority certificate which you used to sign the ssl certificate with to your android? Thus you can open the webdav server on your android in your browser without getting any security warning? (depending on the browser you use, they might or might not use the android certificate store tho)
What's the exact error you are getting in the floccus app?

**Comment by misterhsp** [comment]
I like to repeat myself again:

Just as a side note: Android has not trusted user certificates for a long time. This is nothing new. I mean, it's been like that since Android 6. Look at the Nextcloud app, there is this option that it works with self-signed. That's what you need to put in your app.

...

**Comment by marcelklehr** [comment]
@misterhsp I believe you when you say it doesn't work for you. My knowledge of Android is limited, so all I can say is, I will investigate, try to reproduce and fix this. Thank you for raising awareness for this!
